cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -743,23 +744,116 @@ public void updateSchedule(HttpRequest request, HttpResponder responder,
     doUpdateSchedule(request, responder, namespaceId, appName, appVersion, scheduleName);
   }
 
-  // TODO: refactor doAddSchedule and doUpdateSchedule to reduce code duplication
   private void doUpdateSchedule(HttpRequest request, HttpResponder responder, String namespaceId, String appId,
```

Comment:
I think there should still be an update API. The semantics of that, for now are to delete and create. But I do think we should keep the API